File: backend/app/models/factura.py
import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, ForeignKey, Index, Numeric, String, Text, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base
from app.models.mixins import BaseModelMixin


class Factura(Base, BaseModelMixin):
    __tablename__ = 'facturas'

    usuario_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey('users.id', ondelete='CASCADE'), nullable=False,
    )
    periodo: Mapped[str] = mapped_column(String(7), nullable=False)
    detalle: Mapped[str] = mapped_column(Text, nullable=False)
    referencia_archivo: Mapped[str | None] = mapped_column(String(500), nullable=True)
    tamano_kb: Mapped[float | None] = mapped_column(Numeric(10, 2), nullable=True)
    estado: Mapped[str] = mapped_column(String(20), nullable=False, default='Pendiente')
    cargada_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), nullable=False,
    )
    abonada_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)

    __table_args__ = (
        Index('ix_facturas_usuario_periodo', 'usuario_id', 'periodo'),
        Index('ix_facturas_estado', 'estado'),
    )

File: trace-app/src/features/finanzas-facturas/types/factura.ts
export interface Factura {
  id: string;
  usuario_id: string;
  periodo: string;
  detalle: string;
  referencia_archivo: string | null;
  tamano_kb: number | null;
  estado: string;
  cargada_at: string;
  abonada_at: string | null;
}

export interface CreateFacturaDTO {
  usuario_id: string;
  periodo: string;
  detalle: string;
  referencia_archivo?: string | null;
  tamano_kb?: number | null;
}

export interface CambioEstadoDTO {
  estado: string;
}
